test/integration/puppetlabs/puppetserver/bootstrap_testutils.clj

```diff
@@ -58,6 +75,41 @@
        ~config
        ~@body)))
 
+(defmacro with-puppetserver-running-with-services-and-mock-jrubies
+  "This macro should be used with caution; it makes tests run much more quickly,
+  but you should be careful to make sure that the mocking won't be subverting
+  any important test coverage.  For this reason, we require a `docstring` argument
+  to be passed in, as a sort of annotation explaining why you feel it's safe to
+  use this mocking in your test."
+  [docstring app services config-overrides & body]
+  (let [config (load-dev-config-with-overrides config-overrides)]
+    `(let [services# (conj ~services
+                           (jruby-puppet-testutils/mock-jruby-pool-manager-service
+                            ~config))]
+       (tk-testutils/with-app-with-config
+        ~app
+        services#
+        ~config
+        ~@body))))
+
+(defmacro with-puppetserver-running-with-services-and-mock-jruby-puppet-fn
```

Comment:
Could this simply be a 2nd arity of the macro immediately above?  (i.e. optional `mock-jruby-puppet-fn` parameter)